lib/src/fileset.rs

```diff
@@ -159,6 +161,47 @@ impl FilePattern {
         Self::file_glob_at(dir, pattern, true)
     }
 
+    /// Pattern that matches files with the given extension recursively from
+    /// cwd.
+    pub fn cwd_extension_glob(
+        path_converter: &RepoPathUiConverter,
+        extension: impl AsRef<str>,
+    ) -> Result<Self, FilePatternParseError> {
+        let ext_glob = Self::format_extension_glob(extension)?;
+        Self::cwd_file_glob(path_converter, ext_glob)
+    }
+
+    /// Pattern that matches files with the given extension recursively from cwd
+    /// (case-insensitive).
+    pub fn cwd_extension_glob_i(
+        path_converter: &RepoPathUiConverter,
+        extension: impl AsRef<str>,
+    ) -> Result<Self, FilePatternParseError> {
+        let ext_glob = Self::format_extension_glob(extension)?;
+        Self::cwd_file_glob_i(path_converter, ext_glob)
+    }
+
+    fn format_extension_glob(extension: impl AsRef<str>) -> Result<String, FilePatternParseError> {
+        let ext = extension.as_ref();
+        // Trim a single leading dot so that both "rs" and ".rs" are accepted.
+        let ext = ext.strip_prefix('.').unwrap_or(ext);
+        Self::validate_extension(ext)?;
+        Ok(format!("**/*.{ext}"))
+    }
+
+    /// Validates that an extension contains only valid filename characters.
+    fn validate_extension(extension: &str) -> Result<(), FilePatternParseError> {
```

Comment:
`globset::escape()` can be used instead. It's not important to support `extension = "?"`, etc., but we don't have to reject it either.